lib/pages/home/pages/search/widgets/top_posts.dart

```diff
@@ -0,0 +1,268 @@
+import 'package:async/async.dart';
+import 'package:flutter/material.dart';
+
+class OBTopPosts extends StatefulWidget {
+  final OBTopPostsController controller;
+
+  OBTopPosts({
+    this.controller,
+  });
+
+  @override
+  State<StatefulWidget> createState() {
+    return OBTopPostsState();
+  }
+}
+
+class OBTopPostsState extends State<OBTopPosts> with AutomaticKeepAliveClientMixin {
+  UserService _userService;
+  LocalizationService _localizationService;
+  NavigationService _navigationService;
+  CancelableOperation _getTrendingPostsOperation;
+  OBPostsStreamController _obPostsStreamController;
+
+  bool _needsBootstrap;
+  List<TopPost> _currentTopPosts = [];
+  List<Post> _currentPosts = [];
+  List<int> _excludedCommunities = [];
+  List<Post> _topPostsCacheData = [];
+  int _topPostLastViewedId;
+
+  @override
+  void initState() {
+    super.initState();
+    _needsBootstrap = true;
+    _obPostsStreamController = OBPostsStreamController();
+    if (widget.controller != null) widget.controller.attach(this);
+  }
+
+  @override
+  void dispose() async {
+    super.dispose();
+    await _userService.setStoredTopPosts(_currentTopPosts);
+    if (_getTrendingPostsOperation != null) _getTrendingPostsOperation.cancel();
+  }
+
+  Future refresh() {
+    return _obPostsStreamController.refreshPosts();
+  }
+
+  void scrollToTop() {
+    _obPostsStreamController.scrollToTop();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    var openbookProvider = OpenbookProvider.of(context);
+    _userService = openbookProvider.userService;
+    _localizationService = openbookProvider.localizationService;
+    _navigationService = openbookProvider.navigationService;
+
+    if (_needsBootstrap) _bootstrap();
+    if (!_needsBootstrap) {
+      return OBPostsStream(
+        streamIdentifier: 'explorePostsTab',
+        refresher: _postsStreamRefresher,
+        onScrollLoader: _postsStreamOnScrollLoader,
+        controller: _obPostsStreamController,
+        isTopPostsStream: true,
+        initialPosts: _currentPosts,
+        refreshOnCreate: _currentPosts.length == 0,
+        postBuilder: _topPostBuilder,
+        prependedItems: <Widget>[
+          Padding(
+            padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 10),
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: <Widget>[
+                OBPrimaryAccentText(
+                    _localizationService.post__top_posts_title,
+                    style: TextStyle(
+                        fontWeight: FontWeight.bold, fontSize: 24)),
+                OBIconButton(
+                  OBIcons.settings,
+                  themeColor: OBIconThemeColor.primaryAccent,
+                  onPressed: _onWantsToSeeExcludedCommunities,
+                )
+              ],
+            ),
+          )
+        ],
+      );
+    } else {
+      return Center(
```

Comment:
Why do we need loading here? The OBPostsStream has a loading already 🤔

Reply:
See above